recording_external_revenue/tests/test_conversion.py

```diff
@@ -0,0 +1,126 @@
+# © 2020 - today Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import pytest
+from datetime import datetime, timedelta
+from ddt import ddt, data
+from odoo.exceptions import ValidationError
+from .common import ExternalRevenueCase
+
+
+@ddt
+class TestConversion(ExternalRevenueCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.company = cls.env.user.company_id
+        cls.r1 = cls._create_raw_revenue()
+        cls.r2 = cls._create_raw_revenue()
+        cls.r3 = cls._create_raw_revenue()
+        cls.raw_revenues = cls.r1 | cls.r2 | cls.r3
+
+    @classmethod
+    def _create_raw_revenue(cls, **kwargs):
+        return cls.env["recording.external.revenue.raw"].create(kwargs)
+
+    @classmethod
+    def _find_jobs(cls, revenues):
+        return (
+            cls.env["queue.job"]
+            .search(
+                [
+                    ("model_name", "=", "recording.external.revenue.raw"),
+                    ("method_name", "=", "convert"),
+                ]
+            )
+            .filtered(lambda j: set(j.record_ids) == set(revenues.ids))
+        )
+
+    @data(
+        "partner", "country", "state",
+    )
+    def test_revenues_group_by(self, field):
+        self.r1[field] = "value 1"
+        self.r2[field] = "value 2"
+        self.r3[field] = "value 1"
+        groups = self.raw_revenues.group_by_common_parameters()
+        groups = sorted(groups, key=lambda g: len(g))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Lambda may not be necessary](https://app.codacy.com/manual/numigi/odoo-recording-addons/pullRequest?prid=5620550)